package/MDAnalysis/analysis/contacts.py

```diff
@@ -614,10 +628,10 @@ def run(self, store=True, force=False, start_frame=1, end_frame=None, step_value
             A, B = self.selections
             # determine the end_frame value to use:
             total_frames = self.universe.trajectory.n_frames
-            if not end_frame:
+            if not stop:
                 # use the total number of frames in trajectory if no final value specified
                 end_frame = total_frames
```

Comment:
This whole thing can be deleted. `None` for `stop` is perfectly fine in the slice and will do exactly what you want.